src/nanoemoji/reorder_glyphs.py

```diff
@@ -0,0 +1,205 @@
+    (ot.ContextPos, 2): [ReorderCoverage()],
+    (ot.ContextPos, 3): [ReorderCoverage()],
+    (ot.ChainContextPos, 1): [ReorderCoverage(parallel_list_attr="ChainPosRuleSet")],
+    (ot.ChainContextPos, 2): [ReorderCoverage()],
+    (ot.ChainContextPos, 3): [
+        ReorderCoverage(coverage_attr="BacktrackCoverage"),
+        ReorderCoverage(coverage_attr="InputCoverage"),
+        ReorderCoverage(coverage_attr="LookAheadCoverage"),
+    ],
+    # GSUB
+    (ot.ContextSubst, 1): [ReorderCoverage(parallel_list_attr="SubRuleSet")],
+    (ot.ContextSubst, 2): [ReorderCoverage()],
+    (ot.ContextSubst, 3): [ReorderCoverage()],
+    (ot.ChainContextSubst, 1): [ReorderCoverage(parallel_list_attr="ChainSubRuleSet")],
+    (ot.ChainContextSubst, 2): [ReorderCoverage()],
+    (ot.ChainContextSubst, 3): [
+        ReorderCoverage(coverage_attr="BacktrackCoverage"),
+        ReorderCoverage(coverage_attr="InputCoverage"),
+        ReorderCoverage(coverage_attr="LookAheadCoverage"),
+    ],
+    (ot.ReverseChainSingleSubst, 1): [
+        ReorderCoverage(parallel_list_attr="Substitute"),
+        ReorderCoverage(coverage_attr="BacktrackCoverage"),
+        ReorderCoverage(coverage_attr="LookAheadCoverage"),
+    ],
+    # GDEF
+    (ot.AttachList, None): [ReorderCoverage(parallel_list_attr="AttachPoint")],
+    (ot.LigCaretList, None): [ReorderCoverage(parallel_list_attr="LigGlyph")],
+    (ot.MarkGlyphSetsDef, None): [ReorderCoverage()],
+    # MATH
+    (ot.MathGlyphInfo, None): [ReorderCoverage(coverage_attr="ExtendedShapeCoverage")],
+    (ot.MathItalicsCorrectionInfo, None): [
+        ReorderCoverage(parallel_list_attr="ItalicsCorrection")
+    ],
+    (ot.MathTopAccentAttachment, None): [
+        ReorderCoverage(
+            coverage_attr="TopAccentCoverage", parallel_list_attr="TopAccentAttachment"
+        )
+    ],
+    (ot.MathKernInfo, None): [
+        ReorderCoverage(
+            coverage_attr="MathKernCoverage", parallel_list_attr="MathKernInfoRecords"
+        )
+    ],
+    (ot.MathVariants, None): [
+        ReorderCoverage(
+            coverage_attr="VertGlyphCoverage",
+            parallel_list_attr="VertGlyphConstruction",
+        ),
+        ReorderCoverage(
+            coverage_attr="HorizGlyphCoverage",
+            parallel_list_attr="HorizGlyphConstruction",
+        ),
+    ],
+}
+
+
+def _access_path(path: SubTablePath):
+    path_parts = []
+    for entry in path:
+        path_part = entry.name
+        if entry.index is not None:
+            path_part += f"[{entry.index}]"
+        path_parts.append(path_part)
+    return ".".join(path_parts)
+
+
+def reorder_glyphs(font: ttLib.TTFont, new_glyph_order: List[str]):
+    # Changing the order of glyphs in a TTFont requires that all tables that use
+    # glyph indexes have been fully.
+    # Cf. https://github.com/fonttools/fonttools/issues/2060
+    require_fully_loaded(font)
+
+    font.setGlyphOrder(new_glyph_order)
```

Comment:
Looking at the range request implementation of glyph reordering: https://github.com/litherum/StreamableFonts/blob/master/Optimizer/ReorderFont.py#L98

That scripts del's a couple of cached copies of the original glyph ordering, do you need to do that as well?

Reply:
I believe not as setGlyphOrder handles both of those issues (https://github.com/fonttools/fonttools/blob/main/Lib/fontTools/ttLib/ttFont.py#L453). I believe it has long handled reverse and the handling of glyf is new.